code401challenges/src/main/java/code401challenges/tree/BinaryTree.java

```diff
@@ -45,6 +47,24 @@
         return list;
     }
 
+//    printLevelOrder
+    public ArrayList<T> breadthFirstTraversal(TreeNode<T> root) {
+        ArrayList<T> list = new ArrayList<>();
+        Queue<TreeNode<T>> queue = new LinkedList<>();
+        queue.add(root);
```

Comment:
Only add the root if it's not null. If this were called on an empty tree it would cause an exception.